Es testing

JarrodBaker opened the pull request:
As discussed with @d0choa , we wanted a way to identify if the FE and BE interface has broken. The FE are extracting their queries into '*.gql' files so that we can use those as test inputs. 

This PR pulls those files from the FE repository and saves them in test/resources, auto generates common inputs for target and disease based on Checkomatic's files, and fires the requests against the API to see what breaks. 

I still need to figure out how to handle 'fragments' which are in stand-alone files, but for most queries it works nicely. It even found a bug in the FE already!

JarrodBaker committed:
Adding tests for GraphQL queries from FE

JarrodBaker committed:
Adding GQL test inputs for common targets/diseases

These were taken from Checkomatic

JarrodBaker committed:
Add test inputs for GraphQL

- Update git ignore to include autogenerated files

JarrodBaker committed:
Refactor QqlTest to use generated inputs.

Each GqlCase now has its own generator as that determines which inputs
are required for the API call. This way we can reuse the same test logic
with a variety of inputs.

Other changes:
  - refactor inputs to be read from file rather than coded
  - use scalacheck to test multitude of inputs against ES
  - Add TargetDisease case class

JarrodBaker committed:
Add documentation regarding GqlTest

JarrodBaker committed:
Add updateGqlFiles task to build.sbt

Improved output regarding which files are new/updated.

JarrodBaker committed:
Add IntegrationTestTag to integration tests.

JarrodBaker committed:
Refactor gqlTest into proper classes.

JarrodBaker committed:
Merge branch 'master' into es-testing

mkarmona commented on build.sbt:
```diff
@@ -65,3 +66,57 @@ libraryDependencies ++= Seq(
   "com.sksamuel.elastic4s" %% "elastic4s-json-play" % s4sVersion
 )
+// Google app engine logging
+libraryDependencies += "com.google.cloud" % "google-cloud-logging-logback" % "0.120.0-alpha"
```
@JarrodBaker we should remove this as it was causing problems in the deployment. the default logging system is enough to be capture by the underlying logging transport

JarrodBaker replied:
Good pickup! Thanks and fixed!

JarrodBaker committed:
Remove google cloud logging dependency

JarrodBaker committed:
Add CancerBiomarkers test

- Updated variable generate for Target to include default size and index parameters. Queries that don't require them will ignore them so it wasn't necessary to add another case class.

cmalangone reviewed (approved):
@JarrodBaker this is really a great job.
probably sbt should be split in different files/commands
There is a glitch running sbt on the command line.
